src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -223,6 +224,51 @@ protected override ObjectTypeConstructor BuildSerializationConstructor()
                 serializationInitializers.Add(new ObjectPropertyInitializer(RawDataField, deserializationParameter, null));
             }
 
+            if (ModelTypeMapping != null)
+            {
+                foreach (var propertyWithSerialization in ModelTypeMapping.GetPropertiesWithSerialization())
+                {
+                    if (serializationInitializers.Exists(i => i.Property.Declaration.Name == propertyWithSerialization.Name))
+                        continue;
+                    var csharpType = BuilderHelpers.GetTypeFromExisting(propertyWithSerialization, typeof(object), MgmtContext.TypeFactory);
+                    var isReadOnly = propertyWithSerialization switch
+                    {
+                        IPropertySymbol propertySymbol => propertySymbol.SetMethod == null,
+                        IFieldSymbol fieldSymbol => fieldSymbol.IsReadOnly,
+                        _ => throw new NotSupportedException($"'{propertyWithSerialization.ContainingType.Name}.{propertyWithSerialization.Name}' must be either field or property.")
+                    };
```

Comment:
I believe this is identical to whatever we have in `ModelTypeProviderField`, we could extract this as a method in `BuilderHelpers`.